### PR TITLE
Layout main function of miner

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description
This PR adds the basic miner as a single thread miner. 

## Motivation and Context
This is to get the miner up and running. PR to convert to futures will follow this one as well a refactor upon the common utilities for the conf file and logging from log4rs

## Types of changes
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentationdf

## Checklist:
* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: applications/tari_testnet_miner/src/lib.rs [hunk 1]
```diff
@@ -1,23 +1 @@
-// Copyright 2019. The Tari Project
```
Comment (CjS77) [inline applications/tari_testnet_miner/src/lib.rs]: Copyright disappeared

File: applications/tari_testnet_miner/src/main.rs [hunk 2]
```diff
@@ -53,18 +58,44 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
 
-    // ToDo miner logic
+    let mut miner = Miner::new(ConsensusRules::current());
+    let mut block = proc_block(response.into_inner());
 
-    // ToDo get private key logic
+    loop {
+        let mut line = String::new();
+        println!("Mining, press c to close and stop");
+        let b1 = std::io::stdin().read_line(&mut line).unwrap();
```
Comment (CjS77) [inline applications/tari_testnet_miner/src/main.rs]: You have to sit here hitting enter to mine blocks?

Let me direct your attention to various `ctrlc` crates on crates.io

File: applications/tari_testnet_miner/src/main.rs [hunk 3]
```diff
@@ -53,18 +58,44 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+        if line == "c".to_string() {
+            break;
+        }
+        let height: u64 = if block.header.height <= 2016 {
```
Comment (CjS77) [inline applications/tari_testnet_miner/src/main.rs]: What's happening here?

File: applications/tari_testnet_miner/src/main.rs (hunk 2 above)
Comment (SWvheerden) [reply applications/tari_testnet_miner/src/main.rs]: Futures version solves this, this was just dumping all functionality into a single thread.

File: applications/tari_testnet_miner/src/lib.rs (hunk 1 above)
Comment (SWvheerden) [reply applications/tari_testnet_miner/src/lib.rs]: This is not right, fixed

File: applications/tari_testnet_miner/src/main.rs (hunk 3 above)
Comment (SWvheerden) [reply applications/tari_testnet_miner/src/main.rs]: Currently I am using the bitcoin metric as a stop gap measure to measure the time difference.  Bitcoin looks back 2016 blocks. That is just a edge case to not look negative back at the block height.

**Pull request merged**